pkg/controller/cache/actuator.go

```diff
@@ -57,65 +73,108 @@ func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, ex *extensionsv
-		if err != nil {
-			return fmt.Errorf("failed to compute provider status: %w", err)
-		}
+	registryStatus := computeProviderStatus(services, registryCaches.CASecretName())
 
-		if err := a.updateProviderStatus(ctx, ex, registryStatus); err != nil {
-			return fmt.Errorf("failed to update Extension status: %w", err)
-		}
+	if err = a.updateProviderStatus(ctx, ex, registryStatus); err != nil {
+		return fmt.Errorf("failed to update Extension status: %w", err)
+	}
+
+	if err = secretsManager.Cleanup(ctx); err != nil {
+		return fmt.Errorf("failed to cleanup secrets: %w", err)
 	}
 
 	return nil
 }
 
 // Delete the Extension resource.
-func (a *actuator) Delete(ctx context.Context, _ logr.Logger, ex *extensionsv1alpha1.Extension) error {
+func (a *actuator) Delete(ctx context.Context, logger logr.Logger, ex *extensionsv1alpha1.Extension) error {
 	namespace := ex.GetNamespace()
+	cluster, err := extensionscontroller.GetCluster(ctx, a.client, namespace)
+	if err != nil {
+		return fmt.Errorf("failed to get cluster: %w", err)
+	}
+
+	secretsManager, err := extensionssecretsmanager.SecretsManagerForCluster(ctx, logger.WithName("secretsmanager"), clock.RealClock{}, a.client, cluster, secrets.ManagerIdentity, nil)
+	if err != nil {
+		return err
+	}
+
+	registryCacheServices := registrycacheservices.New(a.client, namespace, registrycacheservices.Values{})
+	if err := component.OpDestroyAndWait(registryCacheServices).Destroy(ctx); err != nil {
+		return fmt.Errorf("failed to destroy the registry caches component: %w", err)
+	}
 
-	registryCaches := registrycaches.New(a.client, namespace, registrycaches.Values{})
+	registryCaches := registrycaches.NewComponent(a.client, secretsManager, namespace, registrycaches.Values{})
 	if err := component.OpDestroyAndWait(registryCaches).Destroy(ctx); err != nil {
 		return fmt.Errorf("failed to destroy the registry caches component: %w", err)
 	}
 
-	return nil
+	return secretsManager.Cleanup(ctx)
 }
 
 // Restore the Extension resource.
-func (a *actuator) Restore(ctx context.Context, log logr.Logger, ex *extensionsv1alpha1.Extension) error {
-	return a.Reconcile(ctx, log, ex)
+func (a *actuator) Restore(ctx context.Context, logger logr.Logger, ex *extensionsv1alpha1.Extension) error {
+	return a.Reconcile(ctx, logger, ex)
 }
 
 // Migrate the Extension resource.
 func (a *actuator) Migrate(ctx context.Context, _ logr.Logger, ex *extensionsv1alpha1.Extension) error {
 	namespace := ex.GetNamespace()
 
-	registryCaches := registrycaches.New(a.client, namespace, registrycaches.Values{
+	registryCacheServices := registrycacheservices.New(a.client, namespace, registrycacheservices.Values{
+		KeepObjectsOnDestroy: true,
+	})
+	if err := component.OpDestroyAndWait(registryCacheServices).Destroy(ctx); err != nil {
+		return fmt.Errorf("failed to destroy the registry cache services component: %w", err)
+	}
+
+	registryCaches := registrycaches.NewComponent(a.client, nil, namespace, registrycaches.Values{
```

Comment:
Did you test CP migration at all? Isn't this a `nil pointer`?

Reply:
Ah, I see. We call only `Destroy` and the secretsManager is not used for `Destroy`.